pkg/apis/core/annotation_key_constants.go

```diff
@@ -82,4 +82,10 @@ const (
 	//
 	// Not all cloud providers support this annotation, though AWS & GCE do.
 	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
+
+	// EndpointsLastTriggerTime is the annotation key, set for endpoints objects, that represents the
+	// timestamp of the last change that triggered the endpoints object update.
```

Comment:
Good example. 
In this case it will be set to T0. We plan to get T0 from lastTransitionTime in PodCondition (https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions)

Extending your example:

at T3, kube-proxy observed endpoints object change
at T4, network was programmed, e.g. iptables were written 

In such case the In-Cluster Network-Programming-Latency SLI will be defined and exported as T4-T0.

Reply:
OK.  Prefer to err on the side of details here, please.